game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -315,6 +319,57 @@ public BattleRound getBattleRoundState() {
     return BattleRound.of(round, maxRounds);
   }
 
+  @Override
+  public @Nullable Territory queryRetreatTerritory(
+      final BattleState battleState,
+      final IDelegateBridge bridge,
+      final GamePlayer retreatingPlayer,
+      final Collection<Territory> availableTerritories,
+      final String text) {
+    return retreatQuery(
+        battleState, getRemote(retreatingPlayer, bridge), availableTerritories, false, text);
+  }
+
+  private @Nullable Territory retreatQuery(
+      final BattleState battleState,
+      final Player remotePlayer,
+      final Collection<Territory> availableTerritories,
+      final boolean submerge,
+      final String text) {
+    final Territory retreatTo =
+        remotePlayer.retreatQuery(
+            battleState.getBattleId(),
+            submerge,
+            battleState.getBattleSite(),
+            availableTerritories,
+            text);
+    if (retreatTo != null && !availableTerritories.contains(retreatTo)) {
```

Comment:
This error handling was added back in 2005.  So it has been around for a while.  It has been a severe log since 2018.  I haven't seen the error show up in github so it is possible it never happens.

Returning null will just have the retreat not happen.  The game will continue as if the user never had a retreat option.

Reply:
I wonder a bit if it is simply just dead code. Given the seeming remote call, one could picture a badly behaving client that would trigger this.

It is a shame that the diff marks don't show more clearly what is moved vs new.

Reply:
It might be.  Looking at the original commit (https://github.com/triplea-game/triplea/commit/94dcbc456ee416c732facb9e512450d43ee98531), the message mentions that the messaging framework is still buggy.

Also, there was a situation in the original code where the selected territory wasn't part of the possible territories.  This happened when a submerge was offered.  But that code was changed sometime ago so that the possible territories included the submerge territory.

Do you want me to change it to an exception or just drop the code?

Reply:
Probably let's follow up with this, but soon as we have spent time on it and it's better to resolve it sooner than later.
IMO if we are pretty confident it is dead code, then drop it. If we think it's only possible if there is a buggy game client (like the server is okay, but the other person is running some sort of hacked up code), then perhaps the current handling is actually more or less okay.. It's hard to really know what to do with this one.